File: ios/MMCoach/MMCoach/App/BackendConfig.swift
//
//  BackendConfig.swift
//  MMCoach
//
//  Back4App connection configuration and SDK bootstrap.
//

import Foundation
import ParseSwift

/// Back4App connection details and Parse-Swift bootstrap.
///
/// `applicationId` and `clientKey` are the app's public client credentials
/// (the mobile-app equivalent of a Firebase `GoogleService-Info.plist`) --
/// they identify this app to Back4App and are meant to ship inside the
/// client binary. They are **not** secrets. Never add the Master Key or
/// REST API Key here; those must stay server-side only.
enum BackendConfig {
    static let applicationId = "BpVzRIbSiAVGFo5V603geaEA6xlblztllwlIlZOd"
    static let clientKey = "Dpi9vjS6zHNG8meRcU7J0DKAegJWGhNgRPTSqOSc"
    static let serverURL = URL(string: "https://parseapi.back4app.com")!

    /// Configures the Parse-Swift SDK. Call once at app launch before any
    /// `BackendService` call is made.
    static func configureParse() {
        ParseSwift.initialize(applicationId: applicationId,
                               clientKey: clientKey,
                               serverURL: serverURL)
    }
}

File: android/app/src/main/java/dev/benderapps/mmcoach/services/BackendConfig.kt
package dev.benderapps.mmcoach.services

/**
 * Back4App connection details -- the Android counterpart to
 * `App/BackendConfig.swift` in the iOS app. Same Back4App application, so
 * these are the same public client credentials as the iOS client ships
 * (not secrets; the Master Key stays server-side only). Parse SDK
 * bootstrap itself lives in [dev.benderapps.mmcoach.MMCoachApplication].
 */
object BackendConfig {
    const val APPLICATION_ID = "BpVzRIbSiAVGFo5V603geaEA6xlblztllwlIlZOd"
    const val CLIENT_KEY = "Dpi9vjS6zHNG8meRcU7J0DKAegJWGhNgRPTSqOSc"
    const val SERVER_URL = "https://parseapi.back4app.com"
}
